sdk/metricsadvisor/ai-metrics-advisor-rest/src/metricsAdvisorClient.ts

```diff
@@ -0,0 +1,144 @@
+import { ClientOptions } from "@azure-rest/core-client";
```

Comment:
I don't think we need to implement this custom client. We should just export the generated client and if there is custom things needed we'd need to create helper functions similar to the paging and lro helpers. The client in RLCs should always be the one generated.

Reply:
In this case since we need a custom policy for KeyAuthentication, we can do a small wrapper function that looks like your constructor. Just not a whole class

```typescript
export function createClient(...) {
   const client = GeneratedClient(...);
   // Add policy, same thing as you have in constructor
  return client;
}
``